amethyst_core/src/geometry.rs

```diff
@@ -0,0 +1,107 @@
+//!
+//! Geometry helper functionality.
+use nalgebra::{Point3, Vector3};
+
+/// A plane which can be intersected by a ray.
+#[derive(Debug, Copy, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct Plane {
+    /// The forward normal of the plane.
+    pub normal: Vector3<f32>,
+    /// The origin/point of the plane.
+    pub position: Vector3<f32>,
```

Comment:
This plane representation is a bit wasteful and contains redundant data. Planes usually just have a normal vector and a single float value bias (implicit plane).